File: public/docs/_examples/dependency-injection/dart/web/heroes/hero.service.dart
// #docregion
import "package:angular2/core.dart";
import "hero.dart";
import "mock-heroes.dart";
import "../logger.service.dart";

@Injectable()
class HeroService {
  String _user;
  // #docregion internals
  Logger _logger;
  bool _isAuthorized;

  HeroService(this._logger, this._isAuthorized);

  getHeroes() {
    var auth = _isAuthorized ? 'authorized' : 'unauthorized';
    _logger.log('Getting heroes for ${auth} user.');
    return HEROES.where((hero) => _isAuthorized || !hero.isSecret).toList();
  }
// #enddocregion internals
}

File: public/docs/_examples/style-guide/ts/07-01/app/hero.service.ts
// #docregion
import { Injectable } from 'angular2/core';
import { Http, Response } from 'angular2/http';

import { Hero } from './hero';

@Injectable()
export class HeroService {
  getHeroes() {
    return this.http.get('api/heroes')
      .map((response: Response) => <Hero[]>response.json().data)
  }
}
